main.go

```diff
@@ -103,7 +109,7 @@ func main() {
 				"audience": []string{cfg.oidc.audience},
 			}
 		}
-		client = &http.Client{
+		clientFetcher = &http.Client{
 			Transport: &oauth2.Transport{
 				Base:   t,
```

Comment:
I think this line negates the instrumentation of the fetcher client on line 74. To make this work, I think we need to something like
```suggestion
				Base:   clientFetcher.Transport,
```

Reply:
good catch, I also moved the creation of both clients to after `if cfg.observatoriumCA != ""` to make sure that `t.TLSClientConfig` gets filled in case it goes inside this if: https://github.com/observatorium/thanos-rule-syncer/pull/13/commits/522bd99cdf2b186316712b412d65ddd7099fe655